### PR TITLE
Added a FIR filter example

### PR DESCRIPTION
**Pull request by wswongat**
## Description & Motivation
I would like to add an example to demonstrate the ROHD can also act like generator.
This example is based on the Chisel example
Don't forget to **rebase and merge** my commits.

## Related Issue(s)
NA

## Testing
This example only do the build check and added in example_test.dart

## Backwards-compatibility

> Is this a breaking change that will not be backwards-compatible?  If yes, how so?

NO

## Documentation

> Does the change require any updates to documentation?  If so, where?  Are they included?

No

### CODE REVIEW

File: test/example_test.dart [hunk 1]
```diff
@@ -12,6 +12,7 @@ import 'package:test/test.dart';
 
 import '../example/example.dart' as counter;
 import '../example/tree.dart' as tree;
+import '../example/fir_filter.dart' as fir_filter;
 
 void main() {
```
Comment (mkorbel1) [inline test/example_test.dart:17]: Add a teardown like this:
```
  tearDown(() {
    Simulator.reset();
  });
```

**Review by mkorbel1 — CHANGES_REQUESTED** [review]
Awesome example addition, thank you for contributing it!  Added a couple minor comments.

**Comment by wswongat** [comment]
@mkorbel1 Thank you for your comments. I just made a commit.

File: example/fir_filter.dart [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    // Register inputs and outputs of the module in the constructor
+    // Module logic must consume registered inputs and output to registered outputs
+    if (depth < 1) {
+      // Depth Check
+      throw Exception('Depth parameter should more than 1');
+    }
+    // Add input/output port
+    en = addInput('en', en);
+    inputVal = addInput('inputVal', inputVal, width: bitWidth);
+    resetB = addInput('reset', resetB);
```
Comment (mkorbel1) [inline example/fir_filter.dart:31]: Minor, but the name of the reset pin should be updated to 'resetB' as well

**Review by mkorbel1 — CHANGES_REQUESTED** [review]
Updates look good except one minor thing, thank you!

Sorry for the delay in re-reviewing, I was out of town.

File: example/fir_filter.dart (hunk 2 above)
Comment (wswongat) [reply example/fir_filter.dart]: Thank you for reminding me.
All fixed.

**Review by mkorbel1 — APPROVED** [review]

**Pull request merged**